### DISCUSSION TITLE
KeyError: 'TURKEY' when importing Zipline with Python v3.7 and v3.8

### DISCUSSION THREAD

**Discussion by HelmieCapital** [stefan-jansen/zipline-reloaded#115] [Q&A] [answered]
Good day,

There seems to be an inexplicable key error when importing Zipline.

Can someone please explain what the issue is ?

The import suddenly stopped working in my project since last night.

Here is the traceback:

Traceback (most recent call last)
[<ipython-input-1-ab8bff6e5ffb>](https://localhost:8080/#) in <module>()
----> 1 import zipline

9 frames
[/usr/local/lib/python3.7/dist-packages/zipline/__init__.py](https://localhost:8080/#) in <module>()
     27 from .utils.numpy_utils import numpy_version
     28 from .utils.pandas_utils import new_pandas
---> 29 from .utils.run_algo import run_algorithm
     30 
     31 # These need to happen after the other imports.

[/usr/local/lib/python3.7/dist-packages/zipline/utils/run_algo.py](https://localhost:8080/#) in <module>()
     22 from zipline.finance import metrics
     23 from zipline.finance.trading import SimulationParameters
---> 24 from zipline.pipeline.data import USEquityPricing
     25 from zipline.pipeline.loaders import USEquityPricingLoader
     26 

[/usr/local/lib/python3.7/dist-packages/zipline/pipeline/__init__.py](https://localhost:8080/#) in <module>()
----> 1 from .classifiers import Classifier, CustomClassifier
      2 from .domain import Domain
      3 from .factors import Factor, CustomFactor
      4 from .filters import Filter, CustomFilter
      5 from .term import Term, LoadableTerm, ComputableTerm

[/usr/local/lib/python3.7/dist-packages/zipline/pipeline/classifiers/__init__.py](https://localhost:8080/#) in <module>()
----> 1 from .classifier import (
      2     Classifier,
      3     CustomClassifier,
      4     Quantiles,
      5     Everything,

[/usr/local/lib/python3.7/dist-packages/zipline/pipeline/classifiers/classifier.py](https://localhost:8080/#) in <module>()
     20 )
     21 from zipline.pipeline.sentinels import NotSpecified
---> 22 from zipline.pipeline.term import ComputableTerm
     23 from zipline.utils.compat import unicode
     24 from zipline.utils.input_validation import expect_types, expect_dtypes

[/usr/local/lib/python3.7/dist-packages/zipline/pipeline/term.py](https://localhost:8080/#) in <module>()
     43 )
     44 
---> 45 from .domain import Domain, GENERIC, infer_domain
     46 from .downsample_helpers import expect_downsample_frequency
     47 from .sentinels import NotSpecified

[/usr/local/lib/python3.7/dist-packages/zipline/pipeline/domain.py](https://localhost:8080/#) in <module>()
     25 from zipline.utils.calendar_utils import get_calendar
     26 
---> 27 from zipline.country import CountryCode
     28 from zipline.utils.formatting import bulleted_list
     29 from zipline.utils.input_validation import expect_types, optional

[/usr/local/lib/python3.7/dist-packages/zipline/country.py](https://localhost:8080/#) in <module>()
      8 
      9 
---> 10 class CountryCode(object):
     11     """A simple namespace of iso3166 alpha2 country codes."""
     12 

[/usr/local/lib/python3.7/dist-packages/zipline/country.py](https://localhost:8080/#) in CountryCode()
     53     TAIWAN = code("TAIWAN, PROVINCE OF CHINA")
     54     THAILAND = code("THAILAND")
---> 55     TURKEY = code("TURKEY")
     56     UNITED_KINGDOM = code("UNITED KINGDOM OF GREAT BRITAIN AND NORTHERN IRELAND")
     57     UNITED_STATES = code("UNITED STATES OF AMERICA")

[/usr/local/lib/python3.7/dist-packages/zipline/country.py](https://localhost:8080/#) in code(name)
      5 
      6 def code(name):
----> 7     return countries_by_name[name].alpha2
      8 
      9 

KeyError: 'TURKEY'

**Comment by RichardDale** [comment, marked as answer]
Workaround already posted for this - it's not a discussion item, so should probably be removed from "Discussions"

**Reply by HelmieCapital** [reply]
Thank you Rich,

Will remove

**Reply by agcap1** [reply]
hey fam, I'm having a hard time finding the work around. how did you fix it?

**Reply by RichardDale** [reply]
Workarounds mentioned in various places including here:
https://github.com/stefan-jansen/zipline-reloaded/issues/114
https://github.com/stefan-jansen/zipline-reloaded/issues/118

(hint: pin the package version of iso3166=2.0.2)

It is actually fixed in the main code, but not yet a formal Zipline-Reloaded release.  Looking forward to the v2.2.1 (or v2.3?) release.  Thanks for your efforts Stefan & MBounouar.
https://github.com/stefan-jansen/zipline-reloaded/pull/117

**Reply by agcap1** [reply]
Hey thanks for the reply!

I thought zipline was discontinued? or was it only dropped by quantopian and picked up by someone else?
